### PR TITLE
EVG-7247 require confirmation for unexpirable volume deletion

### PR DESCRIPTION
**Pull request by ablack12**

### CODE REVIEW

File: rest/route/host_spawn.go [hunk 1]
```diff
@@ -1013,6 +1013,66 @@ func (h *getVolumesHandler) Run(ctx context.Context) gimlet.Responder {
...
+}
+
+func (h *getVolumeByIDHandler) Run(ctx context.Context) gimlet.Responder {
+	v, err := h.sc.FindVolumeById(h.volumeID)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(err)
+	}
+	if v == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
```
Comment (ybrill) [inline rest/route/host_spawn.go:1052]: maybe this could be a 404 (`http.StatusNotFound`)

File: rest/route/host_spawn.go [hunk 2]
```diff
@@ -1013,6 +1013,66 @@ func (h *getVolumesHandler) Run(ctx context.Context) gimlet.Responder {
...
+	if err = volumeDoc.BuildFromService(v); err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "err converting volume '%s' to API model", v.ID))
+	}
+	// if the volume is attached to a host, also return the host ID and volume device name
+	attachedHost, err := h.sc.FindHostWithVolume(v.ID)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "error querying for host"))
+	}
+	if attachedHost != nil {
+		volumeDoc.HostID = model.ToStringPtr(attachedHost.Id)
```
Comment (ybrill) [inline rest/route/host_spawn.go:1066]: why do we need to replace the volume document's host field?

File: rest/route/host_spawn.go [hunk 3]
```diff
@@ -1013,6 +1013,66 @@ func (h *getVolumesHandler) Run(ctx context.Context) gimlet.Responder {
...
+		for _, attachment := range attachedHost.Volumes {
+			if attachment.VolumeID == v.ID {
+				volumeDoc.DeviceName = model.ToStringPtr(attachment.DeviceName)
```
Comment (ybrill) [inline rest/route/host_spawn.go:1069]: do we need the device name?

File: rest/route/host_spawn_test.go [hunk 4]
```diff
@@ -488,6 +488,53 @@ func TestGetVolumesHandler(t *testing.T) {
 	}
 }
 
+func TestGetVolumeByIDHandler(t *testing.T) {
+	assert.NoError(t, db.ClearCollections(host.Collection, host.VolumesCollection))
```
Comment (ybrill) [inline rest/route/host_spawn_test.go:492]: is it necessary to clear collections?

File: rest/route/host_spawn.go (hunk 2 above)
Comment (ablack12) [reply rest/route/host_spawn.go:1066]: 🤷 I can remove it, I copied the other get volumes route but I think it's a holdover from before volumes had a host field.

File: rest/route/host_spawn.go (hunk 3 above)
Comment (ablack12) [reply rest/route/host_spawn.go:1069]: Not for this specifically, but in the future we may use this route and expect it to be populated completely so I'd rather do it now than introduce bugs.

**Pull request merged**